ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/InMemory.hs

```diff
@@ -434,12 +495,14 @@ defaultReadKeySets f dbReader = runReaderT (runDbReader dbReader) f
 -------------------------------------------------------------------------------}
 
 -- | The ledger state at the tip of the chain
-ledgerDbCurrent :: (forall mk. GetTip (l mk)) => LedgerDB l -> l EmptyMK
-ledgerDbCurrent = either unCheckpoint unCheckpoint . AS.head . ledgerDbCheckpoints
+ledgerDbCurrent :: LedgerDB l -> l EmptyMK
+ledgerDbCurrent = seqLast . dbChangelogStates . ledgerDbChangelog
```

Comment:
@nfrisby Do you think that we should actually check for agreement among the implementations when we are performing a query like this? It would require a `OldVsNew l` constraint that might propagate a bit too much, but otherwise it doesn't sound crazy.

Reply:
It might not be a problem if we temporary alter our `TableStuff` to include the `oldVsNew` method, or even better, we could add these two methods:

```haskell
update :: l ValuesMK -> l TrackingMK -> l ValuesMK
cmp :: l mk -> l mk -> Ordering
```